unstructured/ingest/v2/processes/connectors/pinecone.py

```diff
@@ -0,0 +1,182 @@
+        return {
+            "id": str(uuid.uuid4()),
+            "values": element_dict.pop("embeddings", None),
+            "metadata": {
+                "text": element_dict.pop("text", None),
+                "element_serialized": json.dumps(element_dict),
+                **flatten_dict(
+                    element_dict,
+                    separator="-",
+                    flatten_lists=True,
+                    remove_none=True,
+                ),
+            },
+        }
+
+    def run(
+        self,
+        elements_filepath: Path,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        with open(elements_filepath) as elements_file:
+            elements_contents = json.load(elements_file)
+
+        conformed_elements = [
+            self.conform_dict(element_dict=element) for element in elements_contents
+        ]
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        output_path.parent.mkdir(parents=True, exist_ok=True)
+
+        with open(output_path, "w") as output_file:
+            json.dump(conformed_elements, output_file)
+        return output_path
+
+
+@dataclass
+class PineconeUploader(Uploader):
+    upload_config: PineconeUploaderConfig
+    connection_config: PineconeConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    @DestinationConnectionError.wrap
+    def check_connection(self):
+        _ = self.connection_config.create_index()
+
+    @requires_dependencies(["pinecone"], extras="pinecone")
+    def upsert_batch(self, batch):
+        from pinecone.core.client.exceptions import PineconeApiException
+
+        try:
+            index = self.connection_config.create_index()
+            response = index.upsert(batch)
+        except PineconeApiException as api_error:
+            raise DestinationConnectionError(f"http error: {api_error}") from api_error
+        logger.debug(f"results: {response}")
+
+    def run(self, contents: list[UploadContent], **kwargs: Any) -> None:
+
+        elements_dict = []
+        for content in contents:
+            with open(content.path) as elements_file:
+                elements = json.load(elements_file)
+                elements_dict.extend(elements)
+
+        logger.info(
+            f"writing document batches to destination"
+            f" index named {self.connection_config.index_name}"
+            f" environment named {self.connection_config.environment}"
+            f" with batch size {self.upload_config.batch_size}"
+            f" with {self.upload_config.num_of_processes} (number of) processes"
+        )
+
+        pinecone_batch_size = self.upload_config.batch_size
+
+        if self.upload_config.num_of_processes == 1:
+            for chunk in chunk_generator(elements_dict, pinecone_batch_size):
+                self.upsert_batch(chunk)  # noqa: E203
+
+        else:
+            with mp.Pool(
```

Comment:
Since multi processing is controlled at the ingest controller, I think this can be removed.

Reply:
We actually need this for now since the uploader is given the entire batch of documents to upload right now if it's not async. Need to add support for this at the pipeline level.